glean-core/src/upload/directory.rs

```diff
@@ -19,6 +19,23 @@ use crate::{DELETION_REQUEST_PINGS_DIRECTORY, PENDING_PINGS_DIRECTORY};
 /// this will contain the document_id, path, JSON encoded body of a ping and the persisted headers.
 pub type PingPayload = (String, String, String, Option<HeaderMap>);
 
+/// A struct to hold the result of scanning all pings directories.
+#[derive(Clone, Debug, Default)]
+pub struct PingPayloadsByDirectory {
```

Comment:
Somehow this name irks me, but I also don't have a better name.
Just `PingPayloads` is too confusable with `PingPayload`. Hmmm.
Unless you have another good idea we can keep it though.

Reply:
Irks me too, but I also couldn't think of a better name.